pkg/quarterdeck/middleware/auth_test.go

```diff
@@ -136,10 +137,93 @@ func TestGetAccessToken(t *testing.T) {
 	require.ErrorIs(t, err, middleware.ErrParseBearer)
 
 	// Test correct authorization header
-	c.Request.Header.Set("Authorization", "Bearer JWT")
+	token := "eyJhbGciOiJSUzI1NiIsImtpZCI6IjAxR1g2NDdTOFBDVkJDUEpIWEdKUjI2UE42IiwidHlwIjoiSldUIn0.eyJpc3MiOiJodHRwOi8vMTI3LjAuMC4xIiwiYXVkIjpbImh0dHA6Ly8xMjcuMC4wLjEiXSwiZXhwIjoxNjgwNjE1MzMwLCJuYmYiOjE2ODA2MTE3MzAsImlhdCI6MTY4MDYxMTczMCwianRpIjoiMDFneDY0N3M4cGN2YmNwamh4Z2pzcG04N3AiLCJuYW1lIjoiSm9obiBEb2UiLCJlbWFpbCI6Impkb2VAZXhhbXBsZS5jb20iLCJvcmciOiIxMjMiLCJwcm9qZWN0IjoiYWJjIiwicGVybWlzc2lvbnMiOlsicmVhZDpkYXRhIiwid3JpdGU6ZGF0YSJdfQ.LLb6c2RdACJmoT3IFgJEwfu2_YJMcKgM2bF3ISF41A37gKTOkBaOe-UuTmjgZ7WEcuQ-cVkht0KI_4zqYYctB_WB9481XoNwff5VgFf3xrPdOYxS00YXQnl09RRqt6Fmca8nvd4mXfdO7uvpyNVuCIqNxBPXdSnRhreSoFB1GtFm42sBPAD7vF-MQUmU0c4PTsbiCfhR1_buH0NYEE1QFp3vYcgoiXOJHh9VStmRscqvLB12AQrcs26G9opdTCCORmvR2W3JLJ_hliHyp-d9lhXmCDFyiGkDEhTAUglqwBjqz5SO1UfAThWJO18PvZl4QPhb724oNT82VPh0DMDwfw"
+	c.Request.Header.Set("Authorization", "Bearer "+token)
 	tks, err := middleware.GetAccessToken(c)
 	require.NoError(t, err)
-	require.Equal(t, "JWT", tks)
+	require.Equal(t, token, tks)
+
+	// Test correct authorization header lowercase
+	c.Request.Header.Set("authorization", "bearer "+token)
+	tks, err = middleware.GetAccessToken(c)
+	require.NoError(t, err)
+	require.Equal(t, token, tks)
+}
+
+func TestGetAccessTokenCookie(t *testing.T) {
+	// Test both the GetAccessToken Cookie and the SetAuthTokens handler
+	router := gin.New()
+
+	auth, err := authtest.NewServer()
+	require.NoError(t, err, "could not start authentication server")
+	defer auth.Close()
```

Comment:
Unfortunately the only way to test cookies is to set up TLS servers … so this test is _much_ more heavyweight than the other non-cookie tests.